src/ansys/speos/core/generic/general_methods.py

```diff
@@ -108,3 +114,68 @@ def wrapper(*args, **kwargs):
         return method(*args, **kwargs)
 
     return wrapper
+
+
+def error_no_install(install_loc: Union[Path, str], version: Union[int, str]):
+    """Raise error that installation was not found at a location.
+
+    Parameters
+    ----------
+    install_loc : Union[Path, str]
+        Installation Path
+    version : Union[int, str]
+        Version
+    """
+    if not install_loc:
+        install_loc_msg = ""
+    else:
+        install_loc_msg = f"at {Path(install_loc).parent}"
+    raise FileNotFoundError(
+        f"Ansys Speos RPC server installation not found{install_loc_msg}. "
+        f"Please define AWP_ROOT{version} environment variable"
+    )
+
+
+def retrieve_speos_install_dir(
+    speos_rpc_loc: Optional[Union[Path, str]], version: str = LATEST_VERSION
+):
+    """Retrieve Speos install location based on Path or Environment.
+
+    Parameters
+    ----------
+    speos_rpc_loc : Optional[str, Path]
+        location of Speos rpc executable
+    version : str
+        The Speos server version to run, in the 3 digits format, such as "242".
+        If unspecified, the version will be chosen as
+        ``ansys.speos.core.kernel.client.LATEST_VERSION``.
+
+    """
+    if not speos_rpc_loc:
```

Comment:
speos_rpc_loc is a required str/path, it cannot be None.

Reply:
should be allowed to be none-> default value should be none